core/server/lib/abstract-poller.js

```diff
@@ -0,0 +1,57 @@
+const basicSuccessTransformer = () => ({});
+const basicFailTransformer = error => ({error});
+
+class AbstractPolling {
+    constructor(polledFunction, transformSuccess = basicSuccessTransformer, transformFailure = basicFailTransformer) {
+        this.promise = false;
+
+        this.polledFunction = polledFunction;
+        this.transformSuccess = transformSuccess;
```

Comment:
I don't think this is the best way to handle custom state schemas for the poller. I'm not entirely sure how I think we should do it, but one thought is we add a check in the v2 getState method i.e.

```js
if (asyncImporter.error) {
	asyncImporter.errors = [asyncImporter.error];
	delete asyncImporter.error;
}

if (asyncImporter.lastResult.result) {
	asyncImporter.lastResult.problems = ...;
	delete asyncImporter.lastResult.result;
}

return asyncImporter.state;
```

The only issue is the lack of separation

Reply:
I think the poller should just keep the full result, the id and the state as `lastResult`.
And as soon as the importer asks for the last result, it will serialize it for the output.

Reply:
Sounds good, I guess I was trying to optimize where it's not necessary - this endpoint won't get hit terribly often, and when it does, there shouldn't be much traffic in the first place, so caching the transformation will have (basically) no perf benefit